airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/SecretsRepositoryReader.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.config.persistence;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.commons.lang.Exceptions;
+import io.airbyte.config.ConfigSchema;
+import io.airbyte.config.DestinationConnection;
+import io.airbyte.config.SourceConnection;
+import io.airbyte.config.persistence.split_secrets.SecretsHydrator;
+import io.airbyte.validation.json.JsonValidationException;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class is responsible for fetching both connectors and their secrets (from separate secrets
+ * stores). All methods in this class return secrets! Use it carefully.
+ */
+public class SecretsRepositoryReader {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SecretsRepositoryReader.class);
```

Comment:
Nit: you can use `@Slf4j` to instantiate the logger. The const will be named `log` instead of `LOGGER` unfortunately.